server/msgtrace.go

```diff
@@ -330,12 +330,25 @@ func (c *client) initMsgTrace() *msgTrace {
 		return nil
 	}
 	hdr := c.msgBuf[:c.pa.hdr]
+	var external bool
 	// Do not call c.parseState.getHeader() yet for performance reasons.
 	// We first do a "manual" search of the "send-to" destination's header.
 	// If not present, no need to lift the message headers.
 	td := getHeader(MsgTraceSendTo, hdr)
 	if len(td) <= 0 {
-		return nil
+		// If the NATS trace destination header is not present, look for
+		// the external trace header.
+		trcCtxHdrVal := getHeader(trcCtx, hdr)
+		if len(trcCtxHdrVal) <= 0 {
+			return nil
+		}
+		// See if we need to trigger the race based on the last token
+		// that should be set to "01".
+		tk := bytes.Split(trcCtxHdrVal, stringToBytes("-"))
+		if len(tk) != 4 || len([]byte(tk[3])) != 2 || !bytes.Equal(tk[3], stringToBytes("01")) {
```

Comment:
I'll do some testing but just fyi, looks like the initial implemenation in the server was maybe a bit naive and I based my spec on that.

This isnt just a case of a `01` and if its `01` its traced, this is actually a hex encoded bit field - today there is only one value stored in this bit field so the naive check is ok, but we might need to revisit this in future.  And I'll try to get some example trace headers

Reply:
Ok, thanks. I'll check the spec too and see what needs to be done there.